Tests/BowTests/Syntax/MemoizationTest.swift

```diff
@@ -0,0 +1,48 @@
+import XCTest
+import SwiftCheck
+@testable import Bow
+
+class MemoizationTest: XCTestCase {
+
+    let positiveInts = Int.arbitrary.suchThat { x in x > 0 }
+    
+    func testMemoizedFunctionCachesResults() {
+        var timesCalled = 0
+        func longRunningOperation(_ x : Int) -> Int {
+            timesCalled += 1
+            return x + 1
+        }
+        
+        property("Memoized function is only called once for the same input") <- forAll(Int.arbitrary, self.positiveInts) { (input : Int, times : Int) in
+            timesCalled = 0
+            let memoizedFunction = memoize(longRunningOperation)
+            
+            for _ in 0 ..< times {
```

Comment:
what do you think change it to something more swifty?
`(0..<times).forEach { memoizedFunction(input) }`

Reply:
done!